core/src/main/java/org/elasticsearch/tasks/Task.java

```diff
@@ -132,4 +135,16 @@ public Status getStatus() {
     }
 
     public interface Status extends ToXContent, NamedWriteable {}
+
+    public TaskResult result(DiscoveryNode node, Throwable error) throws IOException {
+        return new TaskResult(taskInfo(node, false), error);
+    }
+
+    public TaskResult result(DiscoveryNode node, ActionResponse response) throws IOException {
+        if (response instanceof ToXContent) {
+            return new TaskResult(taskInfo(node, false), (ToXContent) response);
+        } else {
+            throw new IllegalStateException("response has to implement ToXContent for persistence");
```

Comment:
This just ends up in the log if `wait_for_completion=false`, right? I mean, I want to make sure it ends up in there at ERROR level or something, because I think it is always a bug, right?